### PR TITLE
Update ssh-ldap-helper

### PR DESCRIPTION
**Pull request by dariosusman**
Include debug, add LDAP BIND auth, decode base64 output

### CODE REVIEW

File: ssh-ldap-helper [hunk 1]
```diff
@@ -1,6 +1,17 @@
 #!/bin/bash
+  
+URI=$(grep ^uri /etc/ldap/ldap.conf | awk '{print $2}' | awk -F, '{print $1}')
+BASE=$(grep 'base passwd' /etc/ldap/ldap.conf | awk '{print $3}')
+BINDDN=$(grep ^binddn /etc/ldap/ldap.conf | awk '{print $2}')
+BINDPW=$(grep ^bindpw /etc/ldap/ldap.conf | awk '{print $2}')
 
-URI=$(grep ^uri /etc/ldap/ldap.conf |awk '{print $2}' |awk -F, '{print $1}')
-BASE=$(grep ^base /etc/ldap/ldap.conf |awk '{print $2}')
+DEBUG=0
```
Comment (jaychris) [inline ssh-ldap-helper:8]: I'm not sure a DEBUG statement is required when you can just "set -x"?

File: ssh-ldap-helper [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+if [ $DEBUG -eq 1 ]; then
+  echo URI: ${URI}
+  echo BASE: ${BASE}
+  echo BINDDN: ${BINDDN}
+  echo BINDPW: ${BINDPW}
+  echo INPUT: $1
+fi
 
-ldapsearch -x -H ${URI} -b ${BASE} "(&(objectClass=posixAccount)(uid=$1))" sshPublicKey | sed -n '/^ /{H;d};/sshPublicKey:/x;$g;s/\n *//g;s/sshPublicKey: //gp'
+ldapsearch -x -H ${URI} -D ${BINDDN} -w ${BINDPW} -b ${BASE} "(&(objectClass=posixAccount)(uid=$1))" sshPublicKey | sed -n '/^ /{H;d};/sshPublicKey::/x;$g;s/\n *//g;s/sshPublicKey:: //gp' | base64 -d
```
Comment (jaychris) [inline ssh-ldap-helper:17]: What about the cases that don't require binding (as in the original case)?

File: ssh-ldap-helper [hunk 3]
```diff
@@ -1,6 +1,17 @@
 #!/bin/bash
+  
+URI=$(grep ^uri /etc/ldap/ldap.conf | awk '{print $2}' | awk -F, '{print $1}')
+BASE=$(grep 'base passwd' /etc/ldap/ldap.conf | awk '{print $3}')
```
Comment (jaychris) [inline ssh-ldap-helper:4]: why so specific ('base passwd')?

**Review by jaychris — CHANGES_REQUESTED** [review]
I appreciate your changes, but I think they create a use case that is too specific.  If you want to refactor a bit to make sure the original case is still supported, I'd be willing to merge them in.  If you are solving for a specific use case, you might be better off forking...

**Comment by dariosusman** [comment]
Hey Jay.

I think a fork would be the best idea, because it's for a specific case.
In any case, may be it would be best to document the alteratives?

Thank you for your feedback!

Best regards,
Dario Susman

**Pull request closed without merging**